spec/models/provider_spec.rb

```diff
@@ -70,19 +67,18 @@
               'address3' => provider.address3,
               'address4' => provider.address4,
               'postcode' => provider.postcode,
-              'region_code' => ProviderEnrichment.region_codes[provider.region_code]
+              'region_code' => provider.region_code_before_type_cast
             )
           end
+
           it 'returns address of the provider' do
-            enrichment = build(:provider_enrichment, json_data: { 'email' => Faker::Internet.email,
-            'website' => Faker::Internet.url,
-            'train_with_us' => Faker::Lorem.sentence.to_s,
-            'train_with_disability' => Faker::Lorem.sentence.to_s,
-            'address1' => nil,
-            'address2' => nil,
-            'address3' => nil,
-            'address4' => nil,
-            'postcode' => nil })
+            enrichment = build(:provider_enrichment,
```

Comment:
How about a new context for this to be clear that we test the way the API works?